shaders.go

```diff
@@ -127,7 +128,7 @@ func (s *defaultShader) Pre() {
 }
 
 func (s *defaultShader) Draw(texture *gl.Texture, buffer *gl.Buffer, x, y, rotation float32) {
-	if s.lastTexture != texture {
+	if s.lastTexture != texture || s.lastBuffer != buffer{
```

Comment:
This goes into a separate `if`statement. 

```go
if s.lastBuffer != buffer {
    Gl.BindBuffer(Gl.ARRAY_BUFFER, buffer)
}
``` 

And then that line can be removed this `if` statement.